Doc/reference/datamodel.rst

```diff
@@ -804,15 +809,41 @@ Built-in functions
 A built-in function object is a wrapper around a C function.  Examples of
 built-in functions are :func:`len` and :func:`math.sin` (:mod:`math` is a
 standard built-in module). The number and type of the arguments are
-determined by the C function. Special read-only attributes:
-
-* :attr:`!__doc__` is the function's documentation string, or ``None`` if
-  unavailable. See :attr:`function.__doc__`.
-* :attr:`!__name__` is the function's name. See :attr:`function.__name__`.
-* :attr:`!__self__` is set to ``None`` (but see the next item).
-* :attr:`!__module__` is the name of
-  the module the function was defined in or ``None`` if unavailable.
-  See :attr:`function.__module__`.
+determined by the C function.
+
+Special read-only attributes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. list-table::
+   :header-rows: 1
+
+   * - Attribute
+     - Meaning
+
+   * - .. attribute:: builtin_function.__class__
+     - The function's class.
+
```

Comment:
This attribute is present on all Python objects everywhere, and is already documented as such here: https://docs.python.org/3/library/stdtypes.html#special-attributes. Please revert this addition; there's no need to duplicate that documentation here.

```suggestion
```

Reply:
In fact, I've realized that attribute definition duplication is a problem in this chapter. See `__doc__` and `__name__` attributes, e.g.

I'm thinking about a kind of hierarchy: a section with all common attributes and in each callable type something like "the attributes are those above plus these more: ... ". The content that you highlighted now is a good beginning in this direction. However, Reference Language must be referenced by the Library Reference. Besides, we don't have a reference to the cited content in this chapter.

Also, I found the [attribute definitions in the `inspect` module documentation](https://docs.python.org/3/library/inspect.html#types-and-members). This could be erased and added a refer to the Language Reference.

In my opinion, the Language Reference is more important than other sections. All other sections must refer to the Reference, not the contrary.

So, my suggestion is:

1. Keeping this definition (although the duplication). This helps make explicit the problem.
2. Continue this review in this chapter
3. Discussing a format to avoid duplication inside this chapter (and implement it)
4. Removing duplication outside this chapter (the section you cited and `inspect`, for example) and refer to it

What do you think? Any other ideas?

Reply:
Yes, I've been working towards similar goals to you in my recent PRs #112781, https://github.com/python/cpython/pull/112933, https://github.com/python/cpython/pull/112735, etc. My opinions:
- For attributes that really do exist on all objects everywhere, like `__class__` and `__dict__`, the list in `stdtypes.rst` is useful. We should keep the list for things like that. Maybe the list should be moved out of `stdtypes.rst` and into the datamodel somewhere, but that's a separate question from whether we should keep the list or not, and I think we should keep the list.
- For attributes that don't exist on all objects everywhere, like `__name__` and `__doc__`, we should document them on the specific objects for which they exist in the datamodel, and eventually delete them from the list in `stdtypes.rst`. I've been working towards that in my recent PRs.
- Once all things in the big table in `inspect.rst` are properly documented, we should delete it from `inspect.rst` and point to the datamodel docs. I have it on my to-do list to file PRs to improve documentation for attributes on class objects, module objects and coroutine objects.

Reply:
> * For attributes that really do exist on all objects everywhere, like `__class__` and `__dict__`, the list in `stdtypes.rst` is useful. We should keep the list for things like that. Maybe the list should be moved out of `stdtypes.rst` and into the datamodel somewhere, but that's a separate question from whether we should keep the list or not, and I think we should keep the list.

My suggestion is to put the list on `The standard type hierarchy.` The second paragraph already cited special attributes, so this seems like a good place to show them.

Does it make sense for you?